src/Hashids.php

```diff
@@ -263,7 +265,7 @@ public function decode($hash)
             foreach ($hashArray as $subHash) {
                 $alphabet = $this->shuffle($alphabet, substr($lottery.$this->salt.$alphabet, 0, strlen($alphabet)));
                 $result = $this->unhash($subHash, $alphabet);
-                if ($this->math->greaterThan($result, PHP_INT_MAX)) {
+                if ($this->math->greaterThan($result, (string) PHP_INT_MAX)) {
```

Comment:
This looks weird... why is this type cast needed?

Reply:
Yes, I agree. This is just me testing what works and not. This is still very much work in progress.